indexing_utilities.cpp

```diff
@@ -315,73 +395,103 @@ namespace irods {
+                                            {{ "job_category_tag", unique_key }} );
+                                        ++n_jobs;
+                                    }
+                                //} else { throw std::runtime_error {"Parent tags search mode but index type is not 'metadata'"}; }
+                            }
+                            else if ( ! (is_collection && _index_type == "full_text" )) { // full_text is meaningless for collection objects
+                                schedule_policy_event_for_object(
+                                    policy_name,
+                                    path.string(),
+                                    _user_name,
+                                    EMPTY_RESOURCE_NAME,
+                                    _indexer,
+                                    _index_name,
+                                    _index_type,
+                                    generate_delay_execution_parameters(),
+                                    {},{},{},
+                                    {{ "job_category_tag", unique_key }} );
+
+                                ++n_jobs;
+                            }
                         }
-                    }
-                    catch(const exception& _e) {
-                        rodsLog(
-                            LOG_ERROR,
-                            "failed to find indexing resource (error code=[%ld]) for object [%s]",static_cast<long>(_e.code()),
-                            path.string().c_str());
-                    }
-                    catch (const std::runtime_error & e) {
-                        irods::log(LOG_ERROR,fmt::format("Abort indexing collection: {}",e.what()));
-                        break;
-                    }
-                    if (iter != iter_end) { path = iter->path(); }
-                                     else { break; }
-                } // if collection or data object
-            } // for path
+                        catch(const exception& _e) {
+                            rodsLog(
+                                LOG_ERROR,
+                                "failed to find indexing resource (error code=[%ld]) for object [%s]",static_cast<long>(_e.code()),
+                                path.string().c_str());
+                        }
+                        catch (const std::runtime_error & e) {
+                            irods::log(LOG_ERROR,fmt::format("Abort indexing collection: {}",e.what()));
+                            break;
+                        }
+                        if (iter != iter_end) { path = iter->path(); }
+                                         else { break; }
+                    } // if collection or data object
+                } // for path
+            }
+            catch( const path_format_error & f ) {
+                rodsLog(LOG_ERROR , "path_format_error DWM");
+            }
+            catch( const std::runtime_error & f ) {
```

Comment:
std::runtime_error already caught above

Reply:
Line 494, for reference :)

...unless, of course, we anticipate that `iter->path()` could throw (line 498)

Reply:
Now since the throwing of these path related  exceptions is relocated to  `path_calc.hpp` and all `path_*_error` is caught as std::runtime_error, from which class it is descended.  Log messages are printed out, and we break out of the loop in `indexer::schedule_policy_events_for_collection`.  No std::runtime_error clause duplication exists outside the loop.  I think we are good.